packages/nimble-components/src/tree-view/index.ts

```diff
@@ -53,6 +62,47 @@ export class TreeView extends FoundationTreeView {
         return true;
     }
 
+    /**
+     * @internal
+     */
+    public clearTreeGroupSelection(): void {
+        const currentGroupSelection = this.querySelectorAll<TreeItem>(
+            `[${groupSelectedAttribute}]`
+        );
+        currentGroupSelection?.forEach(treeItem => treeItem.removeAttribute(groupSelectedAttribute));
+    }
+
+    /**
+     * @internal
+     */
+    public setGroupSelectionOnRootParentTreeItem(treeItem: HTMLElement): void {
+        this.clearTreeGroupSelection();
+
+        let currentItem: HTMLElement | null | undefined = treeItem;
+        while (
+            currentItem?.parentElement !== this
+            && currentItem?.parentElement !== null
+        ) {
+            currentItem = currentItem?.parentElement;
+        }
+
+        if (currentItem) {
+            currentItem.setAttribute(groupSelectedAttribute, 'true');
+        }
+    }
+
+    // This prevents the toggling of selected state when a TreeItem is clicked multiple times,
+    // which is what the FAST TreeItem allows
+    private handleSelectedChangeOverride(event: Event): void {
+        const toggledElement = event.target as TreeItem | AnchorTreeItem;
```

Comment:
Perhaps this calls for an `ISelectable` (`Selectable`?) interface (which would extend `HTMLElement`).

Reply:
Done